src/extension/storage.ts

```diff
@@ -24,22 +25,42 @@ export function storePayload({
   storageKey: StorageKey;
   port: chrome.runtime.Port;
 }) {
-  const tab = port.sender?.tab;
-  if (!tab?.id) return;
+  if (!port.sender) return;
+  const { tab, url } = port.sender;
+  if (!tab?.id || !url) return;
+  const origin = new URL(url).origin;
   const key = getKeyForRequest(storageKey, payload);
-  localStorage.setItem(key, JSON.stringify({ tabId: tab.id }));
+  const requestInfo: RequestInfo = {
+    tabId: tab.id,
+    origin,
+  };
+  localStorage.setItem(key, JSON.stringify(requestInfo));
 }
 
-export function getTab(storageKey: StorageKey, request: string): number {
+export function getRequestInfo(storageKey: StorageKey, request: string): RequestInfo | undefined {
   const key = getKeyForRequest(storageKey, request);
   const valueJSON = localStorage.getItem(key);
-  const existing: RequestInfo = valueJSON ? JSON.parse(valueJSON) : undefined;
+  if (!valueJSON) return undefined;
+  const existing: RequestInfo = JSON.parse(valueJSON);
```

Comment:
Could this ever be corrupted?

Reply:
Perhaps by something at the OS level with control over Chrome. Not by any other apps or extensions. If it was corrupted, the worst that could happen is a request fails. But, if a malicious party is able to corrupt this, I imagine there are much worse things that can happen (like modifying extension source code)